File: Web_Vue.Server/ViewModels/Auth/UserInfoViewModel.cs
namespace Web_Vue.Server.ViewModels.Auth;

/// <summary> 使用者資料模型（序列化後存入 JWT Claims） </summary>
public class UserInfoViewModel
{
    /// <summary> 使用者 ID </summary>
    public int ID { get; set; }

    /// <summary> 帳號 </summary>
    public string Account { get; set; } = string.Empty;

    /// <summary> 姓名 </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary> 帳號狀態 </summary>
    public UserProfileStatus Status { get; set; }

    /// <summary> 最後登入日期 </summary>
    public DateTime? LatestLoginDate { get; set; }

    /// <summary> 為暫時密碼 </summary>
    public bool IsTemplatePassword { get; set; }

    /// <summary> 暫時密碼驗證有效期 </summary>
    public DateTime? ValidPasswordExpireTime { get; set; }

    /// <summary> 是否已登入（ID > 0 即表示已登入） </summary>
    public bool IsLogin => ID > 0;

    /// <summary> 所屬角色 ID 清單 </summary>
    public List<int> RoleList { get; set; } = new();

    /// <summary> 是否為管理者 </summary>
    public bool IsAdmin { get; set; }

    /// <summary> 角色是否系統維護中 </summary>
    public bool IsMaintaining { get; set; }
}

File: web_vue.client/types/viewmodel/Auth/UserInfoViewModel.ts
// !! 此檔案由 generate-models.mjs 自動生成，請勿手動修改 !!
// 來源：Web_Vue.Server/ViewModels
import type { UserProfileStatusValue } from '../enum'

/** 使用者資料模型（序列化後存入 JWT Claims） */
export interface UserInfoViewModel {
  /** 使用者 ID */
  ID: number
  /** 帳號 */
  Account: string
  /** 姓名 */
  Name: string
  /** 帳號狀態 */
  Status: UserProfileStatusValue
  /** 最後登入日期 */
  LatestLoginDate: string | null
  /** 為暫時密碼 */
  IsTemplatePassword: boolean
  /** 暫時密碼驗證有效期 */
  ValidPasswordExpireTime: string | null
  /** 所屬角色 ID 清單 */
  RoleList: number[]
  /** 是否為管理者 */
  IsAdmin: boolean
  /** 角色是否系統維護中 */
  IsMaintaining: boolean
}
